test/exe/Main.hs

```diff
@@ -415,6 +414,26 @@ codeLensesTests = testGroup "code lenses"
   [ addSigLensesTests
   ]
 
+watchedFilesTests :: TestTree
+watchedFilesTests = testGroup "watched files"
+  [ testSession "workspace file" $ do
+      _ <- openDoc' "A.hs" "haskell" "module A where"
+      RequestMessage{_params = RegistrationParams (List regs)} <- skipManyTill anyMessage (message @RegisterCapabilityRequest)
+      let watchedFileRegs =
+            [ args | Registration _id WorkspaceDidChangeWatchedFiles args <- regs ]
+      liftIO $ assertBool "watches workspace files" $ not $ null watchedFileRegs
+  , testSession "non workspace file" $ do
```

Comment:
I don’t understand this test. `openDoc'` always sends the `DidChangeWatchedFiles` notification so how does this work?
Even assuming this works, I am not sure this is the right thing to test. What we want to test is that the logic for files outside of the workspace works correctly and still picks up new files being created. So how about we change `openDoc'` to not send the notification for files outside the workspace and we add a variant of https://github.com/digital-asset/ghcide/blob/2d71599faf5e93d4ee848043ddd68f10dd295260/test/exe/Main.hs#L236 that creates the file outside of the workspace?

Reply:
The test is not checking what `openDoc'` does, it is checking what `ghcide` does: it should never subscribe to changes in non workspace files. 
I'm happy to add an additional test to check that files outside the workspace get picked up.